### ISSUE TITLE
Refreshing custom patch view when adding commit files to custom patch

### ISSUE THREAD

**Issue by afhoffman** [jesseduffield/lazygit#3258]
**Describe the bug**
When adding a commit file to a custom patch, the bottom part of the right half displays title "Diff" and is empty. The view updates after hitting an arrow key to navigate to another file. Something is not refreshing properly when files are added to the patch builder.

**To Reproduce**

1. Navigate to the commits context
2. Hit enter to view files in a commit
3. Hit space to add file to custom patch

**Expected behavior**
Immediately after adding a commit file to the patch builder, the view should update as in the "post-refresh" screenshot below.

**Screenshots**
Pre-refresh:
<img width="1098" alt="pre-refresh" src="https://github.com/jesseduffield/lazygit/assets/26228688/e9799b11-53e1-426d-8a61-99eb036a4c90">

Post-refresh:
<img width="1098" alt="post-refresh" src="https://github.com/jesseduffield/lazygit/assets/26228688/05919591-5d6d-46b0-81ff-c03744f469f3">

**Version info:**
This happens on current master (05b97c8c), but does not happen on 0.40.2. I'm able to duplicate on macOS, Windows 11, and Ubuntu 22.04 (WSL).

**Additional context**
Doing a git bisect, it looks like this behavior started after 240948b8. This change on my fork (https://github.com/afhoffman/lazygit/commit/d1f1ed0c28e686f24e655aaf4d2837972a8b2517) seems to fix it but I'm not totally sure why that fix works, and it doesn't feel right.

**Comment by stefanhaller** [comment]
Ah, when I made the 240948b88 commit I did feel uncomfortable about it and wondered what would break. I was actually a bit surprised that nothing broke (apparently)... Thanks for the very clear bug report, and for bisecting it.

BTW, we have a similar problem in the Files view, but it's a bit harder to reproduce. The only reproduction recipe I could come up with so far is this:
- set `git.autoRefresh` to false to make it more deterministic
- modify two files in the same directory (don't stage them)
- select the containing directory in the Files view
- in a different terminal, stage one of the files with the command line
- go back to the lazygit terminal: it refreshes, and shows the unstaged file in the "Unstaged changes" pane, but the "Staged changes" pane is empty. Refreshing manually fixes it.

This also broke with 240948b, so it's likely the same root cause. I'll look into it.
